chef/automate
# Jamie/natural sort library function

### :nut_and_bolt: Description: What code changed, and why?
It was determined that it might be useful if we created a helper function for Normal Sorting, which is what we use in our projects-dropdown component.  This could then be reused across our codebase where necessary.

A sorter file was created in app/helpers/auth with a ChefSorters class to hold any future sorting functions we might want to build.

the internals of the projectsArray function in the projects-dropdown component were replaced to use the ChefSorters.normalSort function.

Functions updated in:
- [x] project-dropdown.component.ts
- [x] api-token-list.component.js
- [x] policy-add-members.component.ts
- [x] policy-list.component.js
- [x] project-list.component.js
- [x] projects-filter.reducer.ts
- [x] roles-list.component.js
- [x] team-add-users.component.js
- [x] team-details.component.js
- [x] team-management.component.js
- [x] user-management.component.js

### :chains: Related Resources
https://github.com/chef/automate/pull/1620

### :+1: Definition of Done
A more universal version of a Normal Sorting function is available to use across the codebase.

### :athletic_shoe: How to Build and Test the Change

Visually -
1. build components/automate-ui-devproxy && start_all_services
2. Navigate to https://a2-dev.test/settings/tokens
3. Create an API Token
4. Navigate to https://a2-dev.test/settings/projects
5. Create several projects
6. Navigate to https://a2-dev.test/settings/tokens/ your-api-token
7. Click on the projects dropdown to see your projects sorted normally.

Testing -
1. run the specfile at app/helpers/auth/sorter.spec.ts

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
<img width="1044" alt="Screen Shot 2019-09-17 at 3 46 26 PM" src="https://user-images.githubusercontent.com/16737484/65085222-61a90500-d962-11e9-863f-4fc454bd12df.png">
